test/Microsoft.ML.CodeAnalyzer.Tests/TypeIsSchemaShapeTest.cs

```diff
@@ -2,94 +2,89 @@
+        public async Task ReturnTypeIsSchemaShape()
         {
-            var analyzer = GetCSharpDiagnosticAnalyzer();
-            var diag = analyzer.SupportedDiagnostics[0];
-            var diagC = analyzer.SupportedDiagnostics[2];
-
-            string p(string i = "") => string.IsNullOrEmpty(i) ? "" : $" of item {i}";
-
             var expected = new DiagnosticResult[] {
-                diag.CreateDiagnosticResult(23, 13, p()),
-                diag.CreateDiagnosticResult(24, 13, p("bad")),
-                diag.CreateDiagnosticResult(30, 13, p("c.Item2")),
-                diag.CreateDiagnosticResult(40, 13, p("listen")),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(23, 13).WithArguments(""),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(24, 13).WithArguments(" of item bad"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(30, 13).WithArguments(" of item c.Item2"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeDiagnostic.Rule).WithLocation(40, 13).WithArguments(" of item listen"),
             };
 
-            VerifyCSharpDiagnostic(Source, expected);
+            var test = new VerifyCS.Test { TestCode = Source };
+            test.ExpectedDiagnostics.AddRange(expected);
+            test.Exclusions &= ~AnalysisExclusions.GeneratedCode;
+            await test.RunAsync();
         }
 
         private static string _srcResourceChained;
         internal static string SourceChained => TestUtils.EnsureSourceLoaded(
             ref _srcResourceChained, "TypeIsSchemaShapeResourceChained.cs");
 
         [Fact]
-        public void ReturnTypeIsSchemaShapeChained()
+        public async Task ReturnTypeIsSchemaShapeChained()
         {
             // This is a somewhat more complex example, where instead of direct usage the user of the API is devising their own
             // function where the shape type is a generic type parameter. In this case, we would ideally like the analysis to get
             // chained out of their function.
-            var analyzer = GetCSharpDiagnosticAnalyzer();
-            var diag = analyzer.SupportedDiagnostics[0];
-            var diagTp = analyzer.SupportedDiagnostics[1];
-
-            string p(string i = "") => string.IsNullOrEmpty(i) ? "" : $" of item {i}";
-
             var expected = new DiagnosticResult[] {
-                diagTp.CreateDiagnosticResult(18, 24, "T"),
-                diagTp.CreateDiagnosticResult(41, 24, "T"),
-                diag.CreateDiagnosticResult(55, 26, p("text")),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeParameterDiagnostic.Rule).WithLocation(18, 24).WithArguments("T"),
+                new DiagnosticResult("CS8205", DiagnosticSeverity.Error).WithLocation(21, 52).WithMessage("Attributes are not allowed on local function parameters or type parameters"),
```

Comment:
📝 This seems like a bug in the test input, but I wasn't sure if you wanted me to go ahead and fix it.